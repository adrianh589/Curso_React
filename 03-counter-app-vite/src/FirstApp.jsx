import PropTypes, {string} from "prop-types"; // Si uso TS no es necesario esto porque ya es estricto TS

const newMessage = {
    message: 'Hola Mundo',
    title: 'Titulo'
};

const suma = (a, b) => a + b;


export const FirstApp = ({ title, subtitle, name }) => {
    return (
        <>
            <h1> {title} </h1>
            <h2> {subtitle} </h2>
            <code> { JSON.stringify(newMessage) } </code>
            <p> Soy un subtitulo </p>
            {name}
        </>
    );
}

// Con este le indicamos a react de que tipo queremos que sean
// las variables que ponemos desde el componente padre
FirstApp.propTypes = {
    title: string,
    subtitle: string.isRequired // Asi indicamos que es obligatorio
}

/* Para evitar tener que poner los props por defecto arriba
* y que se vea feo, lo establecemos mejor aqui para que esté
* más ordenado
* */
FirstApp.defaultProps = {
    title: 'No hay ningún título.',
    subtitle: 'No hay subtitulo.',
    name: 'Adrian Hoyos'
}