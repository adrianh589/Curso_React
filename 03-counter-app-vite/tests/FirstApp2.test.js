import React from 'react';
import {render, screen} from '@testing-library/react';
import {FirstApp} from "../src/FirstApp.jsx";


describe('Pruebas en <FirstApp', () => {

    const title = 'asfd';
    const subtitle = 'Soy un subtitulo';

    test('Debe hacer match con el snapshot', () => {

        // Crea un root para la renderizaci√≥n
        const {container} = render(
            <React.StrictMode>
                <FirstApp title={title} subtitle={'aa'}/>
            </React.StrictMode>
        );
        expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar el mensaje "Hola, soy goku"', () => {
        render(
            <React.StrictMode>
                <FirstApp title={title} subtitle={'aa'}/>
            </React.StrictMode>
        );

        expect(screen.getByText(title)).toBeTruthy();

    });

    test('Debe de mostrar el titulo en un h1', () => {
        render(
            <React.StrictMode>
                <FirstApp title={title} subtitle={'aa'}/>
            </React.StrictMode>
        );
        expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);
    });

    test('Debe de mostrar el subtitulo enviado por props', () => {
        render(
            <React.StrictMode>
                <FirstApp title={title} subtitle={subtitle}/>
            </React.StrictMode>
        );
        expect(screen.getByRole('heading', {level: 2}).innerHTML).toContain(subtitle);
    });
});
