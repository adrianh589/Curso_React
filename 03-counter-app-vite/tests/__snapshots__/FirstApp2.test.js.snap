// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    asfd
     
  </h1>
  <h2>
     
    aa
     
  </h2>
  <code>
     
    {"message":"Hola Mundo","title":"Titulo"}
     
  </code>
  <p>
     Soy un subtitulo 
  </p>
  Adrian Hoyos
</div>
`;

exports[`Pruebas en <FirstApp Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    asfd
     
  </h1>
  <h2>
     
    aa
     
  </h2>
  <code>
     
    {"message":"Hola Mundo","title":"Titulo"}
     
  </code>
  <p>
     Soy un subtitulo 
  </p>
  Adrian Hoyos
</div>
`;
